core/src/main/java/org/mule/runtime/core/internal/streaming/object/iterator/CompositeProducer.java

```diff
@@ -35,7 +33,7 @@
    * @param producers a list of {@link Producer}
    */
   public CompositeProducer(List<Producer<T>> producers) {
-    if (CollectionUtils.isEmpty(producers)) {
+    if (producers == null || producers.isEmpty()) {
```

Comment:
Is this change because of performance? It only avoids a method invocation.

Reply:
more or less... the idea is to eventually remove the dependency on commons-collections. With this, there would only remain some base decorators and some other specific things that must be reviewed case by case in order to finish that.